File: src/lib/portal-role.ts
import type { PortalRole } from '@/lib/auth';


export function backendRoleToPortal(role: string): PortalRole | null {
  switch (role.toUpperCase()) {
    case 'STUDENT':
      return 'student';
    case 'PARENT':
      return 'parent';
    case 'TEACHER':
      return 'teacher';
    default:
      return null;
  }
}


export function canManageGrades(role: PortalRole | undefined, apiCanEdit: boolean): boolean {
  return apiCanEdit && role === 'teacher';
}

export function isParentReadOnly(role: PortalRole | undefined): boolean {
  return role === 'parent';
}

export function defaultPortalPath(_role: PortalRole): string {
  return '/accueil';
}

File: user-portal-mobile/lib/models/portal_role.dart
enum PortalRole { student, parent, teacher }

PortalRole? backendRoleToPortal(String role) {
  switch (role.toUpperCase()) {
    case 'STUDENT':
      return PortalRole.student;
    case 'PARENT':
      return PortalRole.parent;
    case 'TEACHER':
      return PortalRole.teacher;
    default:
      return null;
  }
}

String portalRoleLabel(PortalRole role) {
  switch (role) {
    case PortalRole.student:
      return 'Élève';
    case PortalRole.parent:
      return 'Parent';
    case PortalRole.teacher:
      return 'Enseignant';
  }
}
